src/prefect/__init__.py

```diff
@@ -18,6 +18,15 @@
 
 import prefect.agent
 
+# DEPRECATED - this is to allow backwards-compatible access to Parameters
+# This awkward assignment avoids circular imports in core.task
+# https://github.com/PrefectHQ/prefect/pull/2758
+prefect.core.task.Parameter = Parameter  # type: ignore
```

Comment:
If we wanted to actually deprecate access of `Parameter` in the `prefect.core.task` namespace with a warning, I have a hack I've used in other projects to do so. Not sure what the deprecation policy for this project is.

Reply:
I'd love to hear it - in the past I've used `zope.deprecation` but it's way too heavy to bring in just for this. Let me know and I'll update.

Reply:
Mind if I push to this branch?

Reply:
Done. Nothing tricky was needed here, just a subclass with a warning. In the past I've wanted to raise a warning when users imported from a spot without subclassing which was trickier - in this case a subclass should be fine.